src/utils.c

```diff
@@ -374,3 +374,37 @@ SEXP coerceUtf8IfNeeded(SEXP x) {
   return(ans);
 }
 
+// Convert a character vector into a single string, e.g. for printing error messages
+// adapted from https://stackoverflow.com/a/58163237
+// make sure to *free* the returned string after use
+char *joinCharVec (SEXP x, const char *sep)
```

Comment:
I would prefer not using join word for this kind of operation because we use it commonly in context of R's merge. Maybe just concatenate?

Reply:
Makes sense.  I've renamed to `concatCharVec`.